plugin/trino-prometheus/src/test/java/io/trino/plugin/prometheus/TestPrometheusCaseInsensitiveNameMatching.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.prometheus;
+
+import io.airlift.units.Duration;
+import io.trino.spi.connector.SchemaTableName;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.trino.plugin.prometheus.MetadataUtil.METRIC_CODEC;
+import static io.trino.type.InternalTypeManager.TESTING_TYPE_MANAGER;
+import static java.util.Locale.ENGLISH;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestPrometheusCaseInsensitiveNameMatching
+{
+    private PrometheusHttpServer prometheusHttpServer;
+    private static final String DEFAULT_SCHEMA = "default";
+    private static final String UPPER_CASE_METRIC = "UpperCase-Metric";
+
+    @BeforeClass
+    public void setUp()
+    {
+        prometheusHttpServer = new PrometheusHttpServer();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        if (prometheusHttpServer != null) {
+            prometheusHttpServer.stop();
+        }
+    }
+
+    @Test
+    public void testCaseInsensitiveNameMatchingFalse()
+    {
+        PrometheusConnectorConfig config = new PrometheusConnectorConfig();
+        config.setPrometheusURI(prometheusHttpServer.resolve("/prometheus-data/uppercase-metrics.json"));
+        config.setMaxQueryRangeDuration(new Duration(21, DAYS));
+        config.setQueryChunkSizeDuration(new Duration(1, DAYS));
+        config.setCacheDuration(new Duration(30, SECONDS));
+
+        PrometheusClient client = new PrometheusClient(config, METRIC_CODEC, TESTING_TYPE_MANAGER);
+
+        Set<String> tableNames = client.getTableNames(DEFAULT_SCHEMA);
+        assertEquals(1, tableNames.size());
```

Comment:
It would be better to avoid check collection size using  `assertEquals`. I would recommend using AssertJ.
```java
assertThat(tableNames).hasSize(1);
```

Reply:
Done, thx~